### ISSUE TITLE
[Feature Request] Silence the native-compiler with latest Emacs

### ISSUE THREAD

**Issue by dr-scsi** [seagle0128/doom-modeline#527]
The latest Emacs master (822350d198) complains like this:

> In doom-modeline-def-modeline:
> doom-modeline-core.el:1066:2: Warning: docstring has wrong usage of unescaped single quotes (use \= or different quoting)

when native-compiling doom-modeline.  I think the following patch fixes this:

```
diff --git a/doom-modeline-core.el b/doom-modeline-core.el
index 14e3100..a1ff1ea 100644
--- a/doom-modeline-core.el
+++ b/doom-modeline-core.el
@@ -1070,10 +1070,10 @@ LHS and RHS are lists of symbols of modeline segments defined with
 `doom-modeline-def-segment'.
 
 Example:
-  (doom-modeline-def-modeline 'minimal
-    '(bar matches \" \" buffer-info)
-    '(media-info major-mode))
-  (doom-modeline-set-modeline 'minimal t)"
+  (doom-modeline-def-modeline \\='minimal
+    \\='(bar matches \" \" buffer-info)
+    \\='(media-info major-mode))
+  (doom-modeline-set-modeline \\='minimal t)"
   (let ((sym (intern (format "doom-modeline-format--%s" name)))
         (lhs-forms (doom-modeline--prepare-segments lhs))
         (rhs-forms (doom-modeline--prepare-segments rhs)))
diff --git a/doom-modeline-env.el b/doom-modeline-env.el
index 87c8c50..eaf5e3b 100644
--- a/doom-modeline-env.el
+++ b/doom-modeline-env.el
@@ -69,11 +69,11 @@ Example: \"ruby\"")
 
 (defvar-local doom-modeline-env--command-args nil
   "A list of arguments for the command to extract the version from.
-Example: '(\"--version\") ")
+Example: \\='(\"--version\") ")
 
 (defvar-local doom-modeline-env--parser nil
   "A function that returns version number from a command --version (or similar).
-Example: 'doom-modeline-env--ruby")
+Example: \\='doom-modeline-env--ruby")
 
 
 ;; Functions & Macros
@@ -115,7 +115,7 @@ passes on the information into the CALLBACK.
 Example:
   (doom-modeline-env--get
      \"ruby\"
-     '(\"--version\")
+     \\='(\"--version\")
      (lambda (line)
         (message (doom-modeline-parser--ruby line)))"
   (let ((proc (apply 'start-process
diff --git a/doom-modeline-segments.el b/doom-modeline-segments.el
index 49ab1a2..d0ab0aa 100644
--- a/doom-modeline-segments.el
+++ b/doom-modeline-segments.el
@@ -2164,8 +2164,8 @@ mouse-1: Toggle citre mode"
 (defvar doom-modeline-before-github-fetch-notification-hook nil
   "Hooks before fetching GitHub notifications.
 Example:
-  (add-hook 'doom-modeline-before-github-fetch-notification-hook
-            #'auth-source-pass-enable)")
+  (add-hook \\='doom-modeline-before-github-fetch-notification-hook
+            #\\='auth-source-pass-enable)")
 (defun doom-modeline--github-fetch-notifications ()
   "Fetch GitHub notifications."
   (when (and doom-modeline-github
```

**Comment by dr-scsi** [comment]
Welcome.  I turned the patch above into a PR(#528).  I don't think there is an impact for older releases.  `\=` is available for a long time.

**Comment by seagle0128** [comment]
Closed by #528 .
